### PR TITLE
Geo scanner

### PR DESCRIPTION
**Pull request by arcturus**

### CODE REVIEW

**Comment by arcturus** [comment]
@wilsonpage r?

File: MagnetScannerIOS/NetworkResolver.swift [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import SwiftyJSON
+
+//
+// In charge of fetching the urls associated to the latitude and longitude
+// passed.
+//
+public class NetworkResolver {
+  static let API_END_POINT: String = "https://tengam.org/content/v1/search/beacons/"
+  static let SLUG_BASE_URL: String = "https://tengam.org/"
+  static let RADIUS: Int = 1500
```
Comment (wilsonpage) [inline MagnetScannerIOS/NetworkResolver.swift]: You want it this large?

File: MagnetScannerIOS/NetworkResolver.swift [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    for item in json.arrayValue {
+      if let slug = item["slug"].string {
+        let url: String = "\(SLUG_BASE_URL)\(slug)"
+        result.append(JSON(["url": url]))
+      }
+    }
+    
+    return result
+  }
+
```
Comment (wilsonpage) [inline MagnetScannerIOS/NetworkResolver.swift:51]: NIT: Remove line

**Comment by arcturus** [comment]
This closes #4

File: MagnetScannerIOS/NetworkResolver.swift [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+      if let json: JSON = JSON(response.result.value!) {
+        debugPrint("JSON: \(json)")
+        callback(filterJSON(json))
+      } else {
+        callback(empty)
+      }
+    }
+  }
+  
+  private static func filterJSON(json: JSON) -> Array<JSON> {
```
Comment (wilsonpage) [inline MagnetScannerIOS/NetworkResolver.swift:40]: Can you explain what this function does and why it's necessary?

File: MagnetScannerIOS/ScannerGeo.swift [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+//  Magnet
+//
+//  Created by Francisco Jordano on 17/10/2016.
+//
+
+import Foundation
+import CoreLocation
+import SwiftyJSON
+
+class ScannerGeo: NSObject, CLLocationManagerDelegate, Scanner {
```
Comment (wilsonpage) [inline MagnetScannerIOS/ScannerGeo.swift]: NIT: Can we call this `ScannerGeolocation` for consistency?

File: MagnetScannerIOS/ScannerGeo.swift [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+  let lManager: CLLocationManager = CLLocationManager()
```
Comment (wilsonpage) [inline MagnetScannerIOS/ScannerGeo.swift]: NIT: Unnecessary abbreviation -> `locationManager`.

File: MagnetScannerIOS/ScannerGeo.swift [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+  func start() {
+    startLocationManager()
+  }
+  
+  func stop() {
+    lManager.stopUpdatingLocation()
+  }
+  
+  func startLocationManager() -> Bool {
+    guard CLLocationManager.locationServicesEnabled() else {
```
Comment (wilsonpage) [inline MagnetScannerIOS/ScannerGeo.swift]: Is this if the user has denied location services for our app. Can you comment?

File: MagnetScannerIOS/ScannerGeo.swift [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+    super.init()
+    self.callback = callback
+  }
+  
+  func start() {
+    startLocationManager()
+  }
+  
+  func stop() {
+    lManager.stopUpdatingLocation()
```
Comment (wilsonpage) [inline MagnetScannerIOS/ScannerGeo.swift]: Where's the part where we keep telling it to scan in the background even after `.stop()` has been called?

File: MagnetScannerIOS/ScannerGeo.swift [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+    lManager.desiredAccuracy = kCLLocationAccuracyBest
+    lManager.requestWhenInUseAuthorization()
+    
+    lManager.startUpdatingLocation()
+    initialized  = true;
+    
+    return true;
+  }
+  
+  // Deletegate methods
```
Comment (wilsonpage) [inline MagnetScannerIOS/ScannerGeo.swift]: Typo

**Comment by wilsonpage** [comment]
r+ after comments addressed.

**Comment by wilsonpage** [comment]
Will test in app when magnet-client patch arrives :)

File: MagnetScannerIOS/ScannerGeo.swift (hunk 4 above)
Comment (arcturus) [reply MagnetScannerIOS/ScannerGeo.swift]: Sure

File: MagnetScannerIOS/ScannerGeo.swift (hunk 5 above)
Comment (arcturus) [reply MagnetScannerIOS/ScannerGeo.swift]: Right, i was trying that stupid thing when i was getting crazy.

File: MagnetScannerIOS/NetworkResolver.swift (hunk 3 above)
Comment (arcturus) [reply MagnetScannerIOS/NetworkResolver.swift:40]: Right now is just a way of removing the extra json fields that the url returns in order to return just {'url': 'xxxxx'} as the rest of scanners do.

File: MagnetScannerIOS/NetworkResolver.swift (hunk 1 above)
Comment (arcturus) [reply MagnetScannerIOS/NetworkResolver.swift]: Cause I needed that to see something from my place.

Will reduce it to 50 as the android one.

**Comment by arcturus** [comment]
Comments addressed, will land this and create a tag, so the patch in the client can have the final information.

**Pull request merged**